[dotnet-dev-certs] Report logs from CertificateManager on stdout/stderr

0xced opened the pull request:
In commit 8e1e81ae78e876eecad1b816093f0519682e205d it was mentioned:
>  We can decide to write an EventListener in `dotnet-dev-certs` to write the results to the console output.

This is what is achieved with the new `ReporterEventListener` in order to bring back ASP.NET Core v3 feature parity where the --verbose option prints debug messages to help diagnose potential certificate issues. 

Also, all the methods from the CertificateManagerEventSource have been appropriately adapted as per the [EventAttribute][1] documentation.

[1]: https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventattribute

/cc @javiercn

javiercn commented:
/AzurePipelines run AspNetCore-ci

javiercn reviewed (approved):
Looks great!

This needs some manual validation on our side, but after that is good to go!

javiercn commented:
I think there's a call to `ToLower()` that needs to be replaced with a call to `ToLower(InvariantCulture)`

0xced committed:
[dotnet-dev-certs] Report logs from CertificateManager on stdout/stderr

In commit 8e1e81ae78e876eecad1b816093f0519682e205d it was mentioned:
>  We can decide to write an EventListener in `dotnet-dev-certs` to write the results to the console output.

This is what is achieved with the new `ReporterEventListener` in order to bring back ASP.NET Core v3 feature parity where the --verbose option prints debug messages to help diagnose potential certificate issues. 

Also, all the methods from the CertificateManagerEventSource have been appropriately adapted as per the [EventAttribute][1] documentation. Note that it is not required to test for `IsEnabled()` before calling `WriteEvent()` because the very first thing that all `WriteEvent()` overloads do is to return early if not enabled.

[1]: https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventattribute

0xced commented:
I fixed the calls to `ToLowerInvariant()` instead of `ToLower()` and rebased this pull request on master.

Too bad this did not make the cut for the .NET 5 release because I'm having issues with installing the development certificate on Windows that I can't diagnose anymore. `dotnet dev-certs https --trust` worked, I can see the ASP.NET Core HTTPS development certificate in the *Trusted Root Certification Authorities* but `dotnet dev-certs https --check --verbose` is not verbose at all and just returns `No valid certificate found.` 😞

Let's hope it will at least end up in .NET 6. 🤞

0xced commented:
Freshly rebased on the `main` branch. Could you please have a look, @javiercn? Oh, and also remove the `pending author input` tag please. 😉

javiercn commented:
Nevermind our bot

javiercn commented:
Thanks for the contribution!
